---
title: "Chapter 03: The afforestation of alpine ecosystems by alien trees modifies ecological networks of native plant an their associated microbes. "
author: 
  - name: Gabriel Munoz 
    affiliation: 'Concordia University'
  - name: Tonia De Bellis
    affiliation: 'Concordia University'
  - name: Issac Eckert  
    affiliation: 'University of Amsterdam'
  - name: Martin Nunez 
    affiliation: 'Universidad del Comahue'
  - name: Jean-Philippe Lessard 
    affiliation: 'Concordia University'
keywords:
  - Novel ecosystems
  - Pine afforestation 
  - Pinus contorta
  - Plant-soil microbial networks 
  - Network specialization
bibliography: [references3.bib, references2.bib]
execute: 
  eval: false
format:
  html:
    toc: true
    number-sections: true
editor: visual
abstract: |
  <b> Aim: </b>  <br> <b> Location: </b> Northern Patagonia, Argentina.<br>  <b>Time period:</b>Present. <br> <b> Major taxa:</b> Plants, fungi, bacteria <br> <br><b> Results:</b> 
---

## Introduction

Multitrophic interactions between species in a community are the architecture of ecosystems [@bascompte2007plant; @tylianakis2017ecological; @gilman2010framework ]. Yet, we have limited empirical understanding of the changes that multitrophic interaction networks experience when novel ecosystems emerge [@hobbs2009novel; @seibold2018necessity]. Around the globe, many novel ecosystems are emerging facilitated from human mediated dispersal of invasive species and climate change, and often represent combinations of species and ecological dynamics without historical analogues [@hobbs2009novel; @hobbs2013novel, @thomspon2021responding]. While these novel species combinations emerge from altered process of ecological filtering, dispersal, and colonization [@petitpierre2016will; @hobbs2013novel; @alexander2016plant]; novel ecological dynamics can emerge from the reshuffling of species and species interactions in a multitrophic community [@eckert2023single; @pauchard2009ain; @nelson2019food]. Gaining comprehensive understanding of these novel ecological dynamics is important to understand multitrophic biodiversity in a changing world [@thomspon2021responding; @mcdowell2020pervasive; @malhi2020climate].

In terrestrial ecosystems, multitrophic interaction networks between plants and their root-associated microbes regulate nutrient and energy cycling between the above-ground and below-ground inter-phase [@bonfante2009plants,@li2017role; @hassani2018microbial ]. Particularly, the interactions of plants with mutualistic and saprophytic fungi have a crucial role, as they aid plant fungal host in nutrient and water update, provide resistance against pathogens and environmental stress, and promotes the growth of plant biomass [@dong2021phylogenetic, @roy2019processes, @toju2015below; @peay2016mutualistic]. In addition to fungi, plants also interact with Bacteria in the soil [@cong2021coexistence, @dong2022soil; @hassani2018microbial]. Although is yet unclear the role that bacterial play when interacting with plants, previous studies have shown a correlation between the fungal-bacteria diversity ratio of plant-root associated microbial networks with plant resistance to fungal pathogens [@hassani2018microbial; @bever2015maintenance]. Furthermore, simulation and controlled experiments suggest that the topology of plant-root associated networks is sensible to changes in the ecological conditions of an ecosystem[@michalska2022network; @song2020telling], making the topology of plant-root associated microbial network a key indicator of ecosystem structure, stability and multi-functionality in the face of climate change [@woodward2010ecological; \@wu2022network; @cong2021coexistence; @thebault2010stability; @alcantara2012linking; @landi2018complexity]. While previous studies have established a correlation between the topology of plant-soil microbial networks and changes in the soil environment [@dong2022soil; @barberan2012using; @araujo2021distinct; @wu2022network; @ding2022introduction], the ecological processes underpinning multitrophic network assembly remain less well understood [@marjakangas2022trait; @jordan2004network; @michalska2022network], particularly in the context of novel ecosystem formation[@eckert2023single; @rudgers2020climate; @pauchard2009ain]. For instance, we lack comprehensive information on network specialization, which is defined as the degree to which species within a network interact with few or many partners [@bluthgen2007specialization; @bluthgen2006measuring; @armbruster2017specialization], and network turnover, defined as the degree of differences in network composition between sites can inform about the process of network assembly during novel ecosystem formation.

Network turnover be partitioned in two components: turnover resulting from differences between networks in species composition, and turnover resulting from differences between networks in species interactions [@poisot2012dissimilarity], and the correlation of these network turnover components with distinct aspects of network topology, such as network specialization can inform about the underlying network assembly dynamics along a perturbation gradient \[@caradonna2017interaction, @gravel2019bringing; @robroek2021rewiring\]. For instance, in the case of plant-root associated microbial networks during the emergence of novel ecosystems with distinct soil environmental conditions. If species turnover lead to less specialized networks, this may imply fewer constrains for pairwise interaction assembly across trophic levels (e.g. trait mismatches) [@nuismer2013coevolution; @roy2019processes; @peay2016mutualistic]. If interaction turnover (i.e. rewiring) leads to less specialization that may imply an increased hierarchization of the community, where interactions of specialists become subsets of of a small core of highly connected generalists [@toju2015below; @montesinos2012network; @toju2014assembly ]. If species turnover leads to high network specialization, this may indicate that specialists have a selective advantage under changing soil environmental conditions, increasing network compartmentalization, where interactions happen in silos among the network [@dehling2018structure; @nuismer2013coevolution]. If interaction turnover (i.e. rewiring) leads to high specialization that may imply the existence of trade-offs on interaction assembly and strong competition, resulting in little interaction niche overlap [@michalska2022network; @rousk2008examining]. Finally, if network turnover is not related to specialization, this could imply little to none interaction niche differentiation within the study area [@galiana2019geographical; @robinson2020generalists; @zhang2023conversion].

The expansion of non native lodge-pole pine trees (*Pinus contorta*) above the native *Nothophagus* treeline, into previously treeless of the alpine zone in Patagonia is a clear example of the formation of a novel ecosystem [@pauchard2009ain; @mcdougall2011plant]. By increasing shade provisioning, altering soil water regimes and soil chemical composition, the expansion of pine trees to the treeless alpine zone impacts alpine soil abiotic properties such as increasing soil moisture, and reducing available nitrate and phosphorus [@eckert2023single]. In this study, we investigate how plant-fungi and plant-bacteria multitrophic network specialization changes in relation to network turnover along a chronosequence of lodge-pole pine growth. We anticipate that lodge-pole pines being foundational species [@ellison2019foundation] will impact alpine ecosystem function, through their effects on the interaction assembly of native plant-soil microbial networks. In nutrient poor conditions, plants may benefit more from interacting with a wider range of mutualistic fungi than from keeping specialized interactions to reduce competition. Therefore, we expect a reduction in the specialization of plant-soil microbial networks. Furthermore, the reduction of network specialization may buffer local species extinctions promoting robust communities. As such, we also expect network turnover to increase and be mainly driven by interaction rewiring, particularly at the early pine sapling stages, but with an increased relative importance of species turnover as pines grow larger.

## Methods

### Study site

We studied the genesis of a novel ecosystem caused by the ongoing expansion of a non-native tree (Lodgepole pine - *Pinus contorta*) in the alpine ecosystems above the native treelines of Andean mountains in Patagonia, Argentina. Specifically, we investigated one happening at cerro Piltriquitrón, a prominent mountain overlooking the town of El Bolsón, Argentina, latitude 41°59′00″S. Lodgepole pine seeds in this region are dispersed by strong Patagonian winds, which can propel them uphill from pine reservoirs below the treeline at speeds of up to 70 kilometers per hour [@pauchard2009ain; @langdon2010pinus ]. Over the past 20 years, thousands of lodgepole pine trees have successfully taken root, increasing the tree count at the previously treeless alpine plateau [@fig_pat01]. The elevation in our study area spans from *1669* to *1767* meters above sea level. The native vegetation consist on a variety of grasses, shrubs, and forbs well adapted to harsh alpine conditions in Patagonia [@figueroa2022contrasting].  

![**Study design**. We studied a Lodgepole pine (*Pinus contorta*) afforestation in the alpine ecosystems of NW Patagonia, located near the summit of the Piltriquitron mountain and above the native *Nothophaghus* treeline. In there, we established 45 plots divided into 3 treatments: **Control**, **Sapling**, and **Pine**. At each plot, we sampled the roots of native plants and amplify their DNA with fungal and bacterial primers. We build multitrophic networks and study the change in network assembly across the pine afforestation gradient](figs/03_patagonia\Pine_expansion.jpg){#fig-fig_pat01 .lightbox fig-align="center"}

### Experimental design

We focus on sampling the fine roots of native plants in the plots under the influence of of mature pines, pine saplings, and control conditions (i.e., areas with no pines). In total, we established a total of 45 plots, being 15 replicates of 3 treatments. Each plot measured 2x2 meters. We use pine height as a proxy for their age [@franzese2017pine; @garcia2018non ]. The `Pine` plots had adult lodgepole pines taller than 1.5 meters – which by local accounts have been growing on site for the past 15yrs. The `Sapling` plots had pines less than 1 meter in height. `Control` plots were in areas where no pine trees were present. We used an areal image of the study area to delineate a spatial polygon that cover the entire extent of the pine expansion into the alpine plateau. Within this spatial polygon, we selected 15 random points as reference points to set up each of the treatment plots. To remove density dependent effects from an uneven spatial distribution of pine individuals, we ensured there was at least a 10-meter gap between each plot and any nearby pine tree. Additionally, to ensure plot independence, we maintained a minimum separation of 100 meters between plots within the same treatment group and at least 15 meters between plots in the three treatment categories. Fieldwork was conducted from January to March of 2019, corresponding to the Southern Hemisphere's summer season.

### DNA extraction of plant-root associated microbes

We investigated the diversity of soil microorganisms associated with fine roots of native plants at each plot. To do this, we carefully unearthed the root systems of 1-5 plants from each plant species in each plot and placed them in sterile bags for transport to our field laboratory. Between each unearthing, we rigorously cleaned all field equipment with Lysol (Reckitt) to prevent contamination. We also photographed each quadrat and preserved voucher samples of each plant to document the species whose roots we examined. Next, we extracted DNA from the fine roots of these plants following established protocols. This involved rinsing the roots with tap water and storing them in sterile 2ml tubes at -20°C until DNA extraction, which was carried out using the DNeasy PowerSoil Kit (Qiagen). We used specific genetic markers to target different microbial groups: the V5-V6 region of the 16S ribosomal RNA for bacteria, the ITS region for fungi, and we amplified a portion of the small subunit RNA gene for arbuscular mycorrhizal fungi (AMF). This resulted in three distinct microbial communities: (i) root bacteria, (ii) root fungi, and (iii) AMF. For detailed information on DNA extraction, primers, PCR conditions, and Illumina MiSeq sequencing methods, please refer to our previous publication [@eckert2023single]. The Illumina sequences were processed using the DADA2 pipeline [@callahan2016bioconductor; @callahan2016dada2] in `R` to generate a table of amplicon sequence variants (ASVs) [@callahan2016bioconductor] for each microbial group. Taxonomic identity for each ASVs was assigned in DADA2 using the UNITE ver. 8.2 fungal database [@nilsson2019unite] for the fungi, the SILVA 128 database [@glockner201725] for bacteria, and MaarjAM database for AMF [@opik2010online] (details on Bioinformatics, refer to @eckert2023single).

### Measuring soil environmental variation

We measured the soil chemical composition under each treatment. To this end, we collected and pooled soil samples from several random locations within each plot. These samples were then dried and weighed before and after the drying process to calculate soil water content. We sent the dried soil samples to the Departamento de Agronomía at Universidad Nacional Del Sur in Bahía Blanca, Argentina, for analysis of pH, carbon (C), phosphorous (P), nitrate (NO3), and ammonium (NH4) levels.

### Statistical analyses

#### Reconstructing plant-soil microbiome networks

We reconstructed plant-soil microbiome networks only with those ASV observations with read counts exceeding the 10th percentile threshold for each microbial group after applying a natural logarithm transformation. We did this prior to the analyses of the microbial sequence data, to eliminate potentially spurious ASVs caused by PCR error or sequencing artifacts. After, the number of sequences per microbial community was rarefied based on the number of sequences present in the sample with the lowest number sequences to maintain equal sampling depth across samples. Accumulation curves all show a plateau for each rarefied data set, demonstrating that we included most of the microbial diversity in our analyses \[FIGURE ACC CIRVES\]

The three trophic groups examined in this study were saprophytes, pathogens, and mutualists. Following the sample normalization by rarefaction, we generated a list of unique microbial sequences (ASVs) associated with each plant in our study plots for each of the three microbial groups examined. Each ASV was quantified by the number of times it appeared in our rarified data. For all microbial taxa, we included taxonomic information identified at the phylum level. All ASVs not identified to phylum level were removed from the analysis. Furthermore, for the fungi, dataset we employed the FUNGuildR package to assign trophic roles to each ASV and retained only high-confidence records (those identified as *Probable* or *Highly Probable*), discarding those with a *Possible* confidence ranking or missing values.

We standardized sequence read counts by converting them to z-scores for each ASV to facilitate cross-taxa and cross-plant comparisons in our network analysis. This involved subtracting the mean read count for each taxon on each plant from the individual ASV read counts and then dividing by their respective standard deviations. These z-scores were used to construct weighted matrices for each plot, microbial taxon, and plant species, where plant species served as rows and ASVs as columns, with each cell containing the z-score for the corresponding ASV read count.

#### Measuring network turnover

We measured network turnover following the approach of [@poisot2012dissimilarity; @poisot2023guidelines] which separates total network turnover (**β_wn**) into two components related to the dissimilarity of interactions between shared species between network pairs (**β_os**) and the dissimilarity of interactions due to the gain or loss of species between network pairs (**B_st**). We compared the relative importance of these components across our 3 microbial groups and within trophic guilds for the fungi networks. To identify correlations between interaction replacement across microbial domains, we relate **β_wn** between fungi and bacterial networks. We measured the strength of correlations with simple linear models, with **β_wn** of fungi as response variable and **β_wn** of Bacteria as predictor variable. Finally, we measured the homogeneity of species interactions for the set of plant-root associated networks within each treatment. To do this, we measured **β_os’** which is the **β_os** between plot-level networks and the aggregated meta-network for the respective treatment level [@poisot2012dissimilarity; @poisot2023guidelines]

#### Measuring network specialization

We computed network specialization with H2', a metric that varies between 0 to 1. H2' = 0 means that many-to-many interactions dominate the network structure, indicating a full overlap of species' interaction niches. In contrast, H2' = 1 means that species partners are unique between trophic levels, indicating a perfect differentiation of interaction niches. Because networks across plots vary in the number of pairwise interactions, we assessed network specialization using a rarefaction approach [@terry2020finding]. First, we established a sequence of values spanning from 100 to 5000, of lenght 10 and with equally spaced bins. For each value within this sequence, we created subsets of plant-soil microbiome networks by randomly sampling cells from weighted matrices. The probability of selection was based on the ASV read z-scores. Subsequently, we calculated network specialization (H2’), connectance (i.e., ratio of realized vs potential links ), and links per species (i.e., species-level specialization ). This entire process was repeated independently 100 times, and we determined the median of the resulting network metric distribution as our representative network metric for each value in the sequence.

#### Measuring shifts on species network roles

We assessed shifts in the species' network role with the Dbase metric [@toju2017species]. This measure is based on the species' centrality within all plot-level networks, relative to the species' centrality for the meta-web of interactions at the treatment level. We used eigenvector centrality metric as our species level metric of centrality [@bonacich2007some]. Eigenvector centrality is calculated as the set of eigenvectors of the Laplacian matrix representation of the network. We then measured the Dbase metric, which is a metric embedded in the space representing the relationship between centrality of treatment level networks and each of the plot level networks. Dbase measures the deviation from the expected centrality of individual species in present in the treatment level network given a baseline from their expected centrality from local networks. Species with a relatively high Dbase metric, are expected to play a central role in the network at the metacommunity level that is disproportionate to their role in plot-level networks. In other words, rare interactions expressed at the local level, but important for many species at the metacommunity level. [@toju2017species]   

## Results:

### Sampling effort

Our sampling effort yielded 19 native plant species and root-associated microbiome was composed of 5649 Fungal ASVs, 413 AMF- ASVs, and 15138 Bacteria- ASVs. Across all plots, native plants interacted in average with 125 distinct fungal ASVs, about 10 unique AMF-ASVs, and roughly 904 unique bacterial ASVs. Interestingly, for fungi and AMF ASVs, there was significant differences on interaction diversity (i.e., the richness of ASVs among plants within each treatment group). However, this relationship was not observed in bacteria (Table 1). Between treatments, we observed no significant differences regarding the number of distinct fungi ASVs (F = 0.45, p = 0.63) or bacteria (F = 0.05, p = 0.95) per plant. However, the number of distinct AMF ASV per plant (F = 10.10, p = 0.002) differed significantly among the treatments.

### Network turnover

Following our initial expectations, we found a stronger contribution of interaction rewiring (β_os) component relative to the contrition of species turnover (β \_st) to total network turnover (β \_wn) (Fungi: t = 48.87, p \< 0.01, AMF: t = 26.59, p \< 0.01, Bacteria: t = 37.61, p \< 0.01) (Table S1). The interaction rewiring component was relatively more important for networks of AMF of native plants, where the reshaping of interactions between co-occurring species represented a median of 70% of total network turnover. For networks of fungi and bacteria, the rewiring component represented a median of \~55% of total network turnover (Figure 2A). We found that, with a median of 71%, the highest contribution of interaction rewiring to network turnover was for AMF networks between control and sapling treatments. In contrast, with a median of 0.45%, the contributions of the interaction rewiring components to network turnover was the lowest for the differences between bacterial networks within control plots. Indeed, the importance of the interaction component for network differences between and among treatments was generally higher for AMF and lowest for bacteria (Table S2). Within networks of distinct trophic guilds of fungi. we observed that the rewiring component is more important for interactions with mutualists (70%, t = 50.8, p\< 0.001), than for interactions with Saprotrophs (63%, t = 55.4, p\< 0.001) and Pathotrophs (63%, t = 37.5, p\< 0.001) (Figure 2B).

### Effects in network topology: changes in network specialization & species centrality

We found no significant differences between the specialization of networks in pine and sapling treatments relative to control plots for all taxa (Figure 3A). But when comparing the different fungal guilds, we found a significant reduction in the specialization of plant interactions with root associated Pathothrophs, there were not significant differences among treatments in networks with Symbiotroph and Saprotrophs (Table 1). Similarly, we found no significant differences in the heterogeneity of networks across treatments between microbial taxa (Table S4) and between fungal trophic guilds (Table S5). Finally, we discovered that root associated networks under sapling pines had fewer kinds of distinct fungi species that mattered for how the metacommunity was organized compared to control treatments without the presence of alien trees. However, the species roles recovered quickly for networks under the pine trees (Figure 4). A key finding was that species from the order Phacidiales were very important and common pathothroph in fungal networks under saplings. However, for symbiotrophs and saprotrophs the core of relevant taxa was resilient to the effects of alien saplings and adult trees (Figure 4).

Across the landscape of the study area, we found that spatial distance and soil chemistry explain the patterns in network specialization. However, network specialization responded to different soil aspects across distinct taxa. For AM fungal networks changes in network specialization correlate with gradients in soil pH, while for fungal (ITS) networks we found that changes in network specialization respond to changes in available NH4. Finally, for bacterial networks, we found that network specialization changes are related to Phosphorus (Table S3). Across networks between treatments, we found that root associated fungi networks are more specialized than root associated bacterial networks (Figure 3A).