<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Appendix B — Distinct functional responses of consumers and producers to climate drive mutualistic network asymmetry – Assembly Layers Structuring Biodiversity</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../Supplementary_material/S3_historical_biogeography.html" rel="next">
<link href="../Supplementary_material/S1_trait_based_assembly.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-77205cbbef6a5b90ba72edc25dea2156.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../Supplementary_material/S1_trait_based_assembly.html">Appendices</a></li><li class="breadcrumb-item"><a href="../Supplementary_material/S2_functional_asymmetry.html"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">Distinct functional responses of consumers and producers to climate drive mutualistic network asymmetry</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Assembly Layers Structuring Biodiversity</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Manuscript based thesis</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Front matter</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../fm/00_Title_page.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Assembly Layers Structuring Biodiversity</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../03_thesis_abstract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Abstract</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../intro/07_Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/trait_based_assembly/manuscript.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Trait-based inference of ecological network assembly: A conceptual framework and a methodological toolbox</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/functional_asymmetry/manuscript.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Distinct functional responses of consumers and their producers to climate drive mutualistic network asymmetry</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/fossil_turnover/manuscript.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Historical Paleoclimates and the Genesis of Regional Biotas</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../11_general_conclusion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conclusion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Supplementary_material/S1_trait_based_assembly.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Trait-based inference of ecological network assembly</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Supplementary_material/S2_functional_asymmetry.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">Distinct functional responses of consumers and producers to climate drive mutualistic network asymmetry</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Supplementary_material/S3_historical_biogeography.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">C</span>&nbsp; <span class="chapter-title">Climate change drove the deep-time assembly of biodiversity</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#modelling-interaction-assembly" id="toc-modelling-interaction-assembly" class="nav-link active" data-scroll-target="#modelling-interaction-assembly"><span class="header-section-number">B.1</span> Modelling interaction assembly</a>
  <ul class="collapse">
  <li><a href="#stochastic-block-model-sbm" id="toc-stochastic-block-model-sbm" class="nav-link" data-scroll-target="#stochastic-block-model-sbm"><span class="header-section-number">B.1.1</span> Stochastic Block Model (SBM)</a></li>
  <li><a href="#connectance-model" id="toc-connectance-model" class="nav-link" data-scroll-target="#connectance-model"><span class="header-section-number">B.1.2</span> Connectance Model</a></li>
  <li><a href="#matching-trait-model" id="toc-matching-trait-model" class="nav-link" data-scroll-target="#matching-trait-model"><span class="header-section-number">B.1.3</span> Matching-Trait Model</a></li>
  <li><a href="#matching-centrality-model" id="toc-matching-centrality-model" class="nav-link" data-scroll-target="#matching-centrality-model"><span class="header-section-number">B.1.4</span> Matching-Centrality Model</a></li>
  <li><a href="#testing-the-accuracy-of-interaction-assembly-models" id="toc-testing-the-accuracy-of-interaction-assembly-models" class="nav-link" data-scroll-target="#testing-the-accuracy-of-interaction-assembly-models"><span class="header-section-number">B.1.5</span> Testing the accuracy of interaction assembly models</a></li>
  </ul></li>
  <li><a href="#trait-based-downscaling-of-ecological-networks" id="toc-trait-based-downscaling-of-ecological-networks" class="nav-link" data-scroll-target="#trait-based-downscaling-of-ecological-networks"><span class="header-section-number">B.2</span> Trait-based downscaling of ecological networks</a>
  <ul class="collapse">
  <li><a href="#using-sbm-to-cluster-species-into-interaction-guilds" id="toc-using-sbm-to-cluster-species-into-interaction-guilds" class="nav-link" data-scroll-target="#using-sbm-to-cluster-species-into-interaction-guilds"><span class="header-section-number">B.2.1</span> Using SBM to cluster species into interaction guilds</a></li>
  <li><a href="#trait-based-characterization-of-interaction-guilds" id="toc-trait-based-characterization-of-interaction-guilds" class="nav-link" data-scroll-target="#trait-based-characterization-of-interaction-guilds"><span class="header-section-number">B.2.2</span> Trait-based characterization of interaction guilds</a></li>
  <li><a href="#interaction-guild-prediction" id="toc-interaction-guild-prediction" class="nav-link" data-scroll-target="#interaction-guild-prediction"><span class="header-section-number">B.2.3</span> Interaction guild prediction</a></li>
  <li><a href="#building-spatially-explicit-synthetic-networks" id="toc-building-spatially-explicit-synthetic-networks" class="nav-link" data-scroll-target="#building-spatially-explicit-synthetic-networks"><span class="header-section-number">B.2.4</span> Building spatially explicit synthetic networks</a></li>
  </ul></li>
  <li><a href="#evaluating-functional-changes-in-networks-across-environmental-gradients" id="toc-evaluating-functional-changes-in-networks-across-environmental-gradients" class="nav-link" data-scroll-target="#evaluating-functional-changes-in-networks-across-environmental-gradients"><span class="header-section-number">B.3</span> Evaluating functional changes in networks across environmental gradients</a>
  <ul class="collapse">
  <li><a href="#bioclimatic-data" id="toc-bioclimatic-data" class="nav-link" data-scroll-target="#bioclimatic-data"><span class="header-section-number">B.3.1</span> Bioclimatic data</a></li>
  <li><a href="#measuring-interaction-guild-diversity-across-the-environmental-gradient" id="toc-measuring-interaction-guild-diversity-across-the-environmental-gradient" class="nav-link" data-scroll-target="#measuring-interaction-guild-diversity-across-the-environmental-gradient"><span class="header-section-number">B.3.2</span> Measuring interaction guild diversity across the environmental gradient</a></li>
  <li><a href="#measuring-fta" id="toc-measuring-fta" class="nav-link" data-scroll-target="#measuring-fta"><span class="header-section-number">B.3.3</span> Measuring FTA</a></li>
  <li><a href="#does-climate-determine-fta" id="toc-does-climate-determine-fta" class="nav-link" data-scroll-target="#does-climate-determine-fta"><span class="header-section-number">B.3.4</span> Does climate determine FTA?</a></li>
  <li><a href="#does-fta-influence-network-structure" id="toc-does-fta-influence-network-structure" class="nav-link" data-scroll-target="#does-fta-influence-network-structure"><span class="header-section-number">B.3.5</span> Does FTA influence network structure?</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../Supplementary_material/S1_trait_based_assembly.html">Appendices</a></li><li class="breadcrumb-item"><a href="../Supplementary_material/S2_functional_asymmetry.html"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">Distinct functional responses of consumers and producers to climate drive mutualistic network asymmetry</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Appendix B — Distinct functional responses of consumers and producers to climate drive mutualistic network asymmetry</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="modelling-interaction-assembly" class="level2" data-number="B.1">
<h2 data-number="B.1" class="anchored" data-anchor-id="modelling-interaction-assembly"><span class="header-section-number">B.1</span> Modelling interaction assembly</h2>
<p>To capture probabilistic patterns from interactions observed at the continental scale, we tested four modes of interaction assembly: the stochastic block model, the connectance model, the matching-trait model, and the matching-centrality model. (<a href="http://localhost:4649/references.html#ref-terry2020finding">Terry and Lewis 2020</a>). Each model was evaluated based on its ability to estimate interaction probabilities that best predict the observed binary interaction data.</p>
<p>For the trait, connectance, and matching-centrality models, parameters were estimated with the Nelder–Mead algorithm via repeated maximum-likelihood optimization, using multiple random starts and averaging predictions from the best-fitting runs to avoid local optima. The stochastic block model instead assigns species to discrete interaction groups, such that interaction probabilities depend only on group membership, and uses a degree-corrected bipartite SBM algorithm to iteratively improve group assignments based on likelihood gains. To stabilize inference, multiple SBM fits were performed and predictions were averaged across the best solutions, yielding robust estimates of group structure and block-level interaction probabilities.</p>
<section id="stochastic-block-model-sbm" class="level3" data-number="B.1.1">
<h3 data-number="B.1.1" class="anchored" data-anchor-id="stochastic-block-model-sbm"><span class="header-section-number">B.1.1</span> Stochastic Block Model (SBM)</h3>
<p>The SBM assumes that networks exhibit a modular pattern, where subsets of species interact predominantly within other groups of highly connected species (Terry and Lewis, 2020). Interaction probabilities depend only on the group membership of species, not on individual species identities. Species are initially assigned to groups at random, and the model iteratively improves these assignments by moving species between groups to maximize the likelihood of the observed interaction network. This optimization results in an incidence matrix, Z, representing species-level group associations, and a squared matrix, Θ, reflecting interaction probabilities within and between groups. The likelihood (L) of the SBM is given by:</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 1
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[P(a\_{ij} \&gt; 0) = \omega\_{G_i G_j}\]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(a_{ij}\)</span>) denotes the interaction between species (<span class="math inline">\(i\)</span>) and (<span class="math inline">\(j\)</span>); (<span class="math inline">\(G_i\)</span>) and (<span class="math inline">\(G_j\)</span>) are the interaction guild assignments of species (<span class="math inline">\(i\)</span>) and (<span class="math inline">\(j\)</span>); and (<span class="math inline">\(\omega_{G_i G_j}\)</span>) is the estimated probability of interaction between species belonging to those two guilds.</p>
</div>
</div>
</section>
<section id="connectance-model" class="level3" data-number="B.1.2">
<h3 data-number="B.1.2" class="anchored" data-anchor-id="connectance-model"><span class="header-section-number">B.1.2</span> Connectance Model</h3>
<p>In the connectance model species are characterized by continuous connectivity scores that represent their tendency to interact broadly or narrowly within the network. The model assumes a generalist–specialist hierarchy, where interactions of specialist species are probabilistic subsets of those of generalists, rather than arising from discrete groups. Species connectivity scores are optimized by maximizing the likelihood of reproducing the observed interaction matrix. The likelihood (L) of the connectance model is given by:</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 2
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[\operatorname{logit}!\left(P(a\_{ij} \&gt; 0 \mid \mathbf{O})\right) =
c_i + c_j + k_C\]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(a_{ij}\)</span>) denotes the interaction between species (<span class="math inline">\(i\)</span>) and (<span class="math inline">\(j\)</span>); (<span class="math inline">\(c_i\)</span>) and (<span class="math inline">\(c_j\)</span>) are species-specific connectivity parameters; (<span class="math inline">\(k_C\)</span>) is a global intercept controlling overall connectance; and (<span class="math inline">\(\mathbf{O}\)</span>) denotes the observed interaction network.</p>
</div>
</div>
</section>
<section id="matching-trait-model" class="level3" data-number="B.1.3">
<h3 data-number="B.1.3" class="anchored" data-anchor-id="matching-trait-model"><span class="header-section-number">B.1.3</span> Matching-Trait Model</h3>
<p>The matching-trait model assumes that interactions (Pᵢⱼ) are governed by trait compatibility rather than by overall connectance or group membership. Each species is positioned along one or more latent trait axes, and interaction probabilities increase when consumer and resource trait values are closely aligned. Model parameters are optimized by maximizing the likelihood of the observed interaction matrix.</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 3
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[\operatorname{logit}!\left(P(a\_{ij} \&gt; 0 \mid \mathbf{O})\right) =
-\lambda (m_i - m_j)\^2 + k_M\]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(a_{ij}\)</span>) denotes the interaction between species (<span class="math inline">\(i\)</span>) and (<span class="math inline">\(j\)</span>); (<span class="math inline">\(m_i\)</span>) and (<span class="math inline">\(m_j\)</span>) are latent trait values for species (<span class="math inline">\(i\)</span>) and (<span class="math inline">\(j\)</span>); (<span class="math inline">\(\lambda\)</span>) determines how rapidly interaction probability declines with trait mismatch; (<span class="math inline">\(k_M\)</span>) is a global intercept; and (<span class="math inline">\(\mathbf{O}\)</span>) denotes the observed interaction network.</p>
</div>
</div>
</section>
<section id="matching-centrality-model" class="level3" data-number="B.1.4">
<h3 data-number="B.1.4" class="anchored" data-anchor-id="matching-centrality-model"><span class="header-section-number">B.1.4</span> Matching-Centrality Model</h3>
<p>The matching-centrality model integrates approaches from both the connectance and matching-trait models. It simultaneously optimizes species connectivity scores and trait axes to predict interactions. The likelihood (L) for this model is:</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 4
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[\operatorname{logit}!\left(P(a\_{ij} \&gt; 0 \mid \mathbf{O})\right) =
c_i + c_j - \lambda (m_i - m_j)\^2 + k_B \]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(a_{ij}\)</span>) denotes the interaction between species (<span class="math inline">\(i\)</span>) and (<span class="math inline">\(j\)</span>); (<span class="math inline">\(m_i\)</span>) and (<span class="math inline">\(m_j\)</span>) are latent trait positions of species (<span class="math inline">\(i\)</span>) and (<span class="math inline">\(j\)</span>); (<span class="math inline">\(\lambda\)</span>) controls the strength of trait matching; (<span class="math inline">\(c_i\)</span>) and (<span class="math inline">\(c_j\)</span>) represent species-level connectivity effects; (<span class="math inline">\(k_B\)</span>) is a model intercept; and (<span class="math inline">\(\mathbf{O}\)</span>) denotes the observed interaction network.</p>
</div>
</div>
</section>
<section id="testing-the-accuracy-of-interaction-assembly-models" class="level3" data-number="B.1.5">
<h3 data-number="B.1.5" class="anchored" data-anchor-id="testing-the-accuracy-of-interaction-assembly-models"><span class="header-section-number">B.1.5</span> Testing the accuracy of interaction assembly models</h3>
<p>Following guidelines established by Poisot (2023), we evaluated the accuracy of interaction assembly models using Youden’s J statistic.</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 5
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[J = \text{sensitivity} + \text{specificity} - 1\]</span></p>
</div>
</div>
<p>This metric quantifies a model’s ability to correctly distinguish observed interactions from non-interactions by balancing sensitivity (true positive rate) and specificity (true negative rate), and is independent of interaction prevalence. For each model, predicted interaction probabilities were converted to binary predictions using the threshold that maximized Youden’s J, allowing direct comparison of classification performance across alternative assembly hypotheses.In this context, higher values of Youden’s J indicate greater support for a given assembly model in reproducing observed interaction structure beyond random expectation.</p>
</section>
</section>
<section id="trait-based-downscaling-of-ecological-networks" class="level2" data-number="B.2">
<h2 data-number="B.2" class="anchored" data-anchor-id="trait-based-downscaling-of-ecological-networks"><span class="header-section-number">B.2</span> Trait-based downscaling of ecological networks</h2>
<section id="using-sbm-to-cluster-species-into-interaction-guilds" class="level3" data-number="B.2.1">
<h3 data-number="B.2.1" class="anchored" data-anchor-id="using-sbm-to-cluster-species-into-interaction-guilds"><span class="header-section-number">B.2.1</span> Using SBM to cluster species into interaction guilds</h3>
<p>We used the stochastic block model (SBM) as our benchmark latent variable model to capture probabilistic patterns from interactions that are <em>observed</em> at the continental scale.</p>
<p>The output of this model is:</p>
<ul>
<li><p>Incidence matrices reflecting species level associations to a group</p></li>
<li><p>Palms</p></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-331260339.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure B1. Distribution of palm species across interaction guilds.</strong> Bar plot shows the number of palm species assigned to each interaction guild, illustrating differences in species representation among guilds and highlighting guilds with higher or lower palm diversity.</figcaption>
</figure>
</div>
<ul>
<li>Mammals</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-2171192370.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure B2. Distribution of mammal species across interaction guilds.</strong> Bar plot shows the number of mammal species assigned to each interaction guild, highlighting variation in mammal representation among guilds and differences in taxonomic richness across interaction types.</figcaption>
</figure>
</div>
<ul>
<li>a squared matrix reflecting the probability of an interaction between species within and between interaction guilds</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-1469330877.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure B3: Heatmap of pairwise distances among interaction guilds.</strong> Darker red indicates greater strength of interaction between palm and frugivore mammal species-guild combination. Values within heatmap tiles represent the interaction strength expected between species from interaction guild pairs.</figcaption>
</figure>
</div>
</section>
<section id="trait-based-characterization-of-interaction-guilds" class="level3" data-number="B.2.2">
<h3 data-number="B.2.2" class="anchored" data-anchor-id="trait-based-characterization-of-interaction-guilds"><span class="header-section-number">B.2.2</span> Trait-based characterization of interaction guilds</h3>
<p>Characterizing guilds through traits allow interaction patterns to be interpreted in terms of ecological terms rather than topology alone. For example, resource availability is reflected in palm fruit size, stem height, and growth form, which determine which consumers can access and exploit resources. Consumer foraging strategies are captured by mammal diet composition, activity patterns, and body mass, distinguishing guilds dominated by nocturnal generalist frugivores from those structured around smaller, diurnal or crepuscular specialists.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-2838496144.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure B4. Functional differentiation of palms across interaction guilds.</strong> Boxplots show variation in average fruit length and maximum stem height (log-scaled) among interaction guilds, while the heatmap indicates the proportional representation of erect and acaulescent growth forms. Together, these patterns reveal distinct palm trait syndromes associated with different interaction guilds.</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-545621371.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure B5. Functional differentiation of mammals across interaction guilds.</strong> Heatmaps show the proportional composition of mammal diet traits and activity patterns across interaction guilds, while boxplots depict variation in mammalian body mass (log-scaled). Together, these patterns reveal distinct mammal trait syndromes associated with different interaction guilds, linking guild structure to differences in foraging strategy, activity timing, and body size.</figcaption>
</figure>
</div>
</section>
<section id="interaction-guild-prediction" class="level3" data-number="B.2.3">
<h3 data-number="B.2.3" class="anchored" data-anchor-id="interaction-guild-prediction"><span class="header-section-number">B.2.3</span> Interaction guild prediction</h3>
<p>Given that we have information on the trait distribution of the species we have estimated the probability of association with SBM. We use these traits to train multinomial models that predict interaction guilds for all other palm and mammal species in the neotropics without interaction data but information on their traits.</p>
<p>We use multinomial logistic regression models to predict interaction guilds with species trait data. We justify the choice of multinomial logistic regression models as these can handle the prediction of non-binary outcomes, that is in our case, the labeling of SBM groupings per species. We extracted variable importance scores with combinations of the absolute values of the model weights&nbsp;(<a href="http://localhost:4649/references.html#ref-gevrey2003review">Gevrey, Dimopoulos, and Lek 2003</a>).</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-2989354878.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure B6: Trait-based characterization of mammal–palm interaction guilds.</strong> (a) Scatterplot of mammal species body mass (log₁₀ grams) versus percentage of fruit in diet (%), with point shapes indicating diurnal activity (▲ = diurnal, ● = non-diurnal). Ellipses represent trait distributions by interaction guild (IG 1–7). Mammal guilds cluster distinctly along diet and body mass axes, reflecting ecological divergence between small-bodied, highly frugivorous species (e.g.&nbsp;IG-1) and large-bodied, low-frugivory taxa (e.g.&nbsp;IG-6). (b) Palm species characterized by average fruit length (cm) and maximum stem height (m), with ellipses showing trait space by guild and triangle symbols indicating acaulescent forms (1 = acaulescent, 0 = caulescent). Palm guilds occupy contrasting regions of trait space, from tall, small-fruited species (e.g.&nbsp;IG-2) to short, large-fruited, acaulescent types (e.g.&nbsp;IG-6).</figcaption>
</figure>
</div>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Table B1. Trait composition of mammal and palm interaction guilds
</div>
</div>
<div class="callout-body-container callout-body">
<p>Table summarizes species richness and mean trait values for mammal frugivores and palms across interaction guilds 1–7. Values report averages and variability of key functional traits, as well as the proportional representation of activity patterns, diet components, and growth forms, highlighting distinct trait syndromes associated with each guild.</p>
<div class="cell">
<div class="cell-output-display">
<div id="epzpfaxjfv" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#epzpfaxjfv table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#epzpfaxjfv thead, #epzpfaxjfv tbody, #epzpfaxjfv tfoot, #epzpfaxjfv tr, #epzpfaxjfv td, #epzpfaxjfv th {
  border-style: none;
}

#epzpfaxjfv p {
  margin: 0;
  padding: 0;
}

#epzpfaxjfv .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 3px;
  border-top-color: #D3D3D3;
  border-right-style: solid;
  border-right-width: 3px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 3px;
  border-bottom-color: #D3D3D3;
  border-left-style: solid;
  border-left-width: 3px;
  border-left-color: #D3D3D3;
}

#epzpfaxjfv .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#epzpfaxjfv .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#epzpfaxjfv .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#epzpfaxjfv .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#epzpfaxjfv .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#epzpfaxjfv .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#epzpfaxjfv .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#epzpfaxjfv .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#epzpfaxjfv .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#epzpfaxjfv .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#epzpfaxjfv .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#epzpfaxjfv .gt_spanner_row {
  border-bottom-style: hidden;
}

#epzpfaxjfv .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#epzpfaxjfv .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#epzpfaxjfv .gt_from_md > :first-child {
  margin-top: 0;
}

#epzpfaxjfv .gt_from_md > :last-child {
  margin-bottom: 0;
}

#epzpfaxjfv .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#epzpfaxjfv .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#epzpfaxjfv .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#epzpfaxjfv .gt_row_group_first td {
  border-top-width: 2px;
}

#epzpfaxjfv .gt_row_group_first th {
  border-top-width: 2px;
}

#epzpfaxjfv .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#epzpfaxjfv .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#epzpfaxjfv .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#epzpfaxjfv .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#epzpfaxjfv .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#epzpfaxjfv .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#epzpfaxjfv .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#epzpfaxjfv .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#epzpfaxjfv .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#epzpfaxjfv .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#epzpfaxjfv .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#epzpfaxjfv .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#epzpfaxjfv .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#epzpfaxjfv .gt_left {
  text-align: left;
}

#epzpfaxjfv .gt_center {
  text-align: center;
}

#epzpfaxjfv .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#epzpfaxjfv .gt_font_normal {
  font-weight: normal;
}

#epzpfaxjfv .gt_font_bold {
  font-weight: bold;
}

#epzpfaxjfv .gt_font_italic {
  font-style: italic;
}

#epzpfaxjfv .gt_super {
  font-size: 65%;
}

#epzpfaxjfv .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#epzpfaxjfv .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#epzpfaxjfv .gt_indent_1 {
  text-indent: 5px;
}

#epzpfaxjfv .gt_indent_2 {
  text-indent: 10px;
}

#epzpfaxjfv .gt_indent_3 {
  text-indent: 15px;
}

#epzpfaxjfv .gt_indent_4 {
  text-indent: 20px;
}

#epzpfaxjfv .gt_indent_5 {
  text-indent: 25px;
}

#epzpfaxjfv .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#epzpfaxjfv div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<colgroup>
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="gt_heading header">
<th colspan="8" class="gt_heading gt_title gt_font_normal"><strong>Interaction guild</strong></th>
</tr>
<tr class="gt_heading even">
<th colspan="8" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border">Trait summaries by guild (1–7)</th>
</tr>
<tr class="gt_col_headings gt_spanner_row header">
<th rowspan="2" id="a::stub" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col"></th>
<th colspan="7" id="Interaction guild" class="gt_center gt_columns_top_border gt_column_spanner_outer" data-quarto-table-cell-role="th" scope="colgroup"><div class="gt_column_spanner">
Interaction guild
</div></th>
</tr>
<tr class="gt_col_headings even">
<th id="a1" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">1</th>
<th id="a2" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">2</th>
<th id="a3" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">3</th>
<th id="a4" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">4</th>
<th id="a5" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">5</th>
<th id="a6" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">6</th>
<th id="a7" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">7</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="gt_group_heading_row odd">
<td colspan="8" id="Mammal frugivores" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Mammal frugivores</td>
</tr>
<tr class="gt_row_group_first even">
<td id="stub_1_1" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row"># species</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_1 1">66.00</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_1 2">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_1 3">341.00</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_1 4">33.00</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_1 5">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_1 6">15.00</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_1 7">33.00</td>
</tr>
<tr class="odd">
<td id="stub_1_2" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Body mass (mean)</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_2 1">6.47</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_2 2">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_2 3">4.75</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_2 4">3.91</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_2 5">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_2 6">9.73</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_2 7">8.15</td>
</tr>
<tr class="even">
<td id="stub_1_3" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Percentage fruit in diet (mean)</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_3 1">64.09</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_3 2">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_3 3">21.70</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_3 4">48.48</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_3 5">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_3 6">20.00</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_3 7">36.36</td>
</tr>
<tr class="odd">
<td id="stub_1_4" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Body mass (sd)</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_4 1">1.68</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_4 2">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_4 3">1.33</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_4 4">0.66</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_4 5">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_4 6">0.73</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_4 7">0.52</td>
</tr>
<tr class="even">
<td id="stub_1_5" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Percentage fruit in diet (sd)</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_5 1">11.63</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_5 2">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_5 3">7.20</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_5 4">3.64</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_5 5">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_5 6">12.54</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_5 7">4.89</td>
</tr>
<tr class="odd">
<td id="stub_1_6" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Percentage diurnal</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_6 1">48.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_6 2">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_6 3">22.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_6 4">3.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_6 5">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_6 6">40.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_6 7">79.0</td>
</tr>
<tr class="even">
<td id="stub_1_7" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Percentage nocturnal</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_7 1">52.0</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_7 2">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_7 3">84.0</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_7 4">97.0</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_7 5">NA</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_7 6">73.0</td>
<td class="gt_row gt_center" headers="Mammal frugivores stub_1_7 7">24.0</td>
</tr>
<tr class="odd">
<td id="stub_1_8" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Percentage crepuscular</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_8 1">8.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_8 2">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_8 3">10.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_8 4">0.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_8 5">NA</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_8 6">47.0</td>
<td class="gt_row gt_center gt_striped" headers="Mammal frugivores stub_1_8 7">30.0</td>
</tr>
<tr class="gt_group_heading_row even">
<td colspan="8" id="Palms" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Palms</td>
</tr>
<tr class="gt_row_group_first odd">
<td id="stub_1_9" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row"># species</td>
<td class="gt_row gt_center" headers="Palms stub_1_9 1">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_9 2">25.00</td>
<td class="gt_row gt_center" headers="Palms stub_1_9 3">15.00</td>
<td class="gt_row gt_center" headers="Palms stub_1_9 4">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_9 5">265.00</td>
<td class="gt_row gt_center" headers="Palms stub_1_9 6">137.00</td>
<td class="gt_row gt_center" headers="Palms stub_1_9 7">NA</td>
</tr>
<tr class="even">
<td id="stub_1_10" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Stem height (mean)</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_10 1">NA</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_10 2">3.05</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_10 3">3.45</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_10 4">NA</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_10 5">1.95</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_10 6">1.18</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_10 7">NA</td>
</tr>
<tr class="odd">
<td id="stub_1_11" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Fruit length (mean)</td>
<td class="gt_row gt_center" headers="Palms stub_1_11 1">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_11 2">0.80</td>
<td class="gt_row gt_center" headers="Palms stub_1_11 3">1.69</td>
<td class="gt_row gt_center" headers="Palms stub_1_11 4">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_11 5">0.90</td>
<td class="gt_row gt_center" headers="Palms stub_1_11 6">1.39</td>
<td class="gt_row gt_center" headers="Palms stub_1_11 7">NA</td>
</tr>
<tr class="even">
<td id="stub_1_12" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Stem height (sd)</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_12 1">NA</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_12 2">0.27</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_12 3">0.26</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_12 4">NA</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_12 5">0.63</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_12 6">0.96</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_12 7">NA</td>
</tr>
<tr class="odd">
<td id="stub_1_13" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Fruit length (sd)</td>
<td class="gt_row gt_center" headers="Palms stub_1_13 1">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_13 2">0.11</td>
<td class="gt_row gt_center" headers="Palms stub_1_13 3">0.40</td>
<td class="gt_row gt_center" headers="Palms stub_1_13 4">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_13 5">0.31</td>
<td class="gt_row gt_center" headers="Palms stub_1_13 6">0.57</td>
<td class="gt_row gt_center" headers="Palms stub_1_13 7">NA</td>
</tr>
<tr class="even">
<td id="stub_1_14" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Percentage acaulescent</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_14 1">NA</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_14 2">0.0</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_14 3">0.0</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_14 4">NA</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_14 5">1.0</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_14 6">36.0</td>
<td class="gt_row gt_center gt_striped" headers="Palms stub_1_14 7">NA</td>
</tr>
<tr class="odd">
<td id="stub_1_15" class="gt_row gt_left gt_stub" style="text-align: left;" data-quarto-table-cell-role="th" scope="row">Percentage erect</td>
<td class="gt_row gt_center" headers="Palms stub_1_15 1">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_15 2">100.0</td>
<td class="gt_row gt_center" headers="Palms stub_1_15 3">93.0</td>
<td class="gt_row gt_center" headers="Palms stub_1_15 4">NA</td>
<td class="gt_row gt_center" headers="Palms stub_1_15 5">98.0</td>
<td class="gt_row gt_center" headers="Palms stub_1_15 6">64.0</td>
<td class="gt_row gt_center" headers="Palms stub_1_15 7">NA</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</div>
</div>
<section id="important-traits" class="level4" data-number="B.2.3.1">
<h4 data-number="B.2.3.1" class="anchored" data-anchor-id="important-traits"><span class="header-section-number">B.2.3.1</span> Important traits</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-2422444127.png" class="img-fluid figure-img"></p>
<figcaption>Figure B7. Variable importance from neural network models fitted with caret. Bar plots show relative variable importance for the palm model (left) and the mammal model (right), calculated using caret::varImp applied to refitted neural network classifiers. For neural networks, variable importance follows the method of Gevrey et al.&nbsp;(2003), in which importance is derived from combinations of the absolute values of connection weights linking each input variable to the hidden and output layers. For classification models, class specific importances are identical and are therefore shown as a single score per variable. Importance values are scaled relative to the most influential predictor within each model, enabling comparison across traits. Results indicate that plant growth form strongly dominates the palm model, whereas mammal activity patterns are the primary drivers of the mammal model, with secondary contributions from diet and body mass.</figcaption>
</figure>
</div>
</section>
</section>
<section id="building-spatially-explicit-synthetic-networks" class="level3" data-number="B.2.4">
<h3 data-number="B.2.4" class="anchored" data-anchor-id="building-spatially-explicit-synthetic-networks"><span class="header-section-number">B.2.4</span> Building spatially explicit synthetic networks</h3>
<section id="finding-co-occuring-assemblages-at-the-grid-cell-level" class="level4" data-number="B.2.4.1">
<h4 data-number="B.2.4.1" class="anchored" data-anchor-id="finding-co-occuring-assemblages-at-the-grid-cell-level"><span class="header-section-number">B.2.4.1</span> Finding co-occuring assemblages at the grid-cell level</h4>
<p>To identify local co-occurring assemblages of palms and mammals, we converted species range polygons into presence-by-grid-cell representations using a common spatial grid across the Neotropics. When creating grids, we do not consider spherical geometries to avoid topology artifacts (e.g., duplicate vertices) that can occur when intersecting complex polygons as the spatial representations of species ranges. We then created a 1° × 1° grid covering the Neotropics and intersected the grid with each palm and mammal ranges to create <em>local</em> species occurrences. These gridded palm and mammal presence layers can then be combined per grid cell to define local palm assemblages, local mammal assemblages, and ultimately co-occurring palm–mammal assemblages used in downstream network construction and trait-based analyses.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-1532803991.png" class="img-fluid figure-img"></p>
<figcaption>Figure B8. Spatial grid used to define local assemblages across the Neotropics. Map shows the 1° × 1° spatial grid overlaid on the Neotropical region. Grid cells provide the spatial units for intersecting species range polygons and defining local palm and mammal assemblages used in downstream co-occurrence and interaction network analyses.</figcaption>
</figure>
</div>
</section>
<section id="assembling-grid-cell-level-networks" class="level4" data-number="B.2.4.2">
<h4 data-number="B.2.4.2" class="anchored" data-anchor-id="assembling-grid-cell-level-networks"><span class="header-section-number">B.2.4.2</span> Assembling grid-cell level networks</h4>
<p>We use a composite interaction probability score that integrates block-model interaction propensity, relative spatial overlap, and geographic proximity, such that interactions are favored between species assigned to compatible guilds, with overlapping ranges and short spatial separation between their range centroids.</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 6
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[s_{ij} = \left(\Omega_{G_i G_j}\right)\cdot \left(\frac{A_iA_j}{(A_i + A_j)^2}\right)\cdot\left(1 - \tilde d_{ij}\right)\]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(s_{ij}\)</span>) is the final interaction score between palms (<span class="math inline">\(i\)</span>) and mammals (<span class="math inline">\(j\)</span>); (<span class="math inline">\(\Omega_{G_i G_j}\)</span>) is the stochastic block model interaction probability between their guilds; (<span class="math inline">\(A_i\)</span>) and (<span class="math inline">\(A_j\)</span>) are summed range areas; (<span class="math inline">\(d_{ij}\)</span>) is the geographic distance between centroids; (<span class="math inline">\(\tilde d_{ij}\)</span>) is the min–max rescaled distance.</p>
</div>
</div>
<p>This approach allowed us to recreate synthetic probabilistic plant-mammal frugivore networks for each grid-cell across the Neotropics, while accounting for the heterogeneity of species ranges within each grid. We applied an exclusion criteria to the list of grids before applying this network generating algorithm, and selected only those grids with at least 5 palm and 5 mammal species.</p>
</section>
</section>
</section>
<section id="evaluating-functional-changes-in-networks-across-environmental-gradients" class="level2" data-number="B.3">
<h2 data-number="B.3" class="anchored" data-anchor-id="evaluating-functional-changes-in-networks-across-environmental-gradients"><span class="header-section-number">B.3</span> Evaluating functional changes in networks across environmental gradients</h2>
<section id="bioclimatic-data" class="level3" data-number="B.3.1">
<h3 data-number="B.3.1" class="anchored" data-anchor-id="bioclimatic-data"><span class="header-section-number">B.3.1</span> Bioclimatic data</h3>
<p>WorldClim bioclimatic variables are derived from long-term monthly climate surfaces based on weather station records distributed globally. WorldClim variables are generated by spatially interpolating station data using thin-plate splines that incorporate elevation and other covariates, producing continuous gridded climate surfaces that represent regional climatic gradients across complex tropical topography. We used the following as spatial predictors:</p>
<ul>
<li><p><strong>Mean Annual Temperature (BIO01)</strong> represents the average temperature over all months of the year and summarizes the overall thermal energy available to ecosystems.</p></li>
<li><p><strong>Total Annual Precipitation (BIO12)</strong> is the sum of monthly precipitation values and captures the total annual water input to a region.</p></li>
<li><p><strong>Temperature Seasonality (BIO14)</strong> quantifies the amount of temperature variation over the year, calculated as the standard deviation of monthly mean temperatures multiplied by 100, and reflects the degree of thermal variability.</p></li>
<li><p><strong>Precipitation Seasonality (BIO15)</strong> measures the variability of monthly precipitation totals, expressed as the coefficient of variation, and indicates how evenly precipitation is distributed throughout the year.</p></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-2118027858.png" class="img-fluid figure-img"></p>
<figcaption>Figure B9. Climatic context of the Neotropics used in network analyses. Spatial distributions of mean annual temperature, total annual precipitation, precipitation seasonality, and temperature seasonality across the Neotropics. Maps summarize the broad climatic gradients that structure species distributions and interaction assemblages, and serve as environmental predictors for assessing how climate modulates functional richness and interaction guild organization.</figcaption>
</figure>
</div>
</section>
<section id="measuring-interaction-guild-diversity-across-the-environmental-gradient" class="level3" data-number="B.3.2">
<h3 data-number="B.3.2" class="anchored" data-anchor-id="measuring-interaction-guild-diversity-across-the-environmental-gradient"><span class="header-section-number">B.3.2</span> Measuring interaction guild diversity across the environmental gradient</h3>
<p>We performed spatial joins to associate each grid-cell species assemblage with corresponding bioclimatic variables. We then quantified variation in the relative occupancy of interaction guilds inferred by the stochastic block model by calculating, for each grid cell, the proportion of palm or mammal species assigned to each interaction guild. For a given assemblage, we first counted the number of palm species and mammal species assigned to each interaction guild. These counts were then normalized by the total number of species in each trophic level to obtain proportional guild representations.</p>
<p>Unlike trait-space–based measures of functional richness that quantify the volume of multivariate trait space occupied by species, our network-based formulation captures diversity in realized interaction roles rather than diversity in trait values per se. Because interactions in bipartite networks reflect both trait matching and network structure, functional richness defined from SBM-derived interaction guilds captures realized ecological roles rather than only trait diversity.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-288217907.png" class="img-fluid figure-img"></p>
<figcaption>Figure B10: Changes in the functional richness of mammals along climatic gradients across interaction guilds. Each row represents a distinct mammal–palm interaction guild (IG-1 to IG-7), identified through stochastic block modeling, while each column corresponds to an environmental gradient: mean annual temperature (MAT, in °C), total annual precipitation (TAP, in mm), temperature seasonality (TS, expressed as the standard deviation of monthly temperature ×100), and precipitation seasonality (PS, represented as the coefficient of variation). In each panel, the y-axis shows the functional richness of mammal communities, and the x-axis shows the corresponding environmental gradient. Red smoothed lines represent observed trends in mammal functional richness, while green smoothed lines represent trends for the associated palm communities within the same guild. Shaded ribbons around the smoothed lines indicate 95% confidence intervals. Background colors delineate different interaction guilds to aid visual interpretation.</figcaption>
</figure>
</div>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Table B2: Model estimates between FR and climate.
</div>
</div>
<div class="callout-body-container callout-body">
<p>Parametric terms from separate GAMs predicting functional richness (FR) in palms (“FR Palms”) and mammal frugivores (“FR Mammals”). The leftmost column lists the model predictors: the intercept; four climatic main effects—Mean Annual Temperature (MAT), Total Annual Precipitation (TAP), Temperature Seasonality (TS), and Precipitation Seasonality (PS); and interaction terms between each climate variable and the seven interaction guilds (IG-1 through IG-7). Columns “Estimates,” “Std. Error,” and “p” give the coefficient, its standard error, and significance for each predictor in the palm and mammal models, respectively. “IG-n” refers to the nᵗʰ palm–mammal guild as defined in the reference guild table. The bottom rows report the number of grid‐cell observations and the model R² (or adjusted R² for mammals).</p>
<div class="cell">
<div class="cell-output-display">
<div id="tyxwoadtyz" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#tyxwoadtyz table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#tyxwoadtyz thead, #tyxwoadtyz tbody, #tyxwoadtyz tfoot, #tyxwoadtyz tr, #tyxwoadtyz td, #tyxwoadtyz th {
  border-style: none;
}

#tyxwoadtyz p {
  margin: 0;
  padding: 0;
}

#tyxwoadtyz .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 3px;
  border-top-color: #D3D3D3;
  border-right-style: solid;
  border-right-width: 3px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 3px;
  border-bottom-color: #D3D3D3;
  border-left-style: solid;
  border-left-width: 3px;
  border-left-color: #D3D3D3;
}

#tyxwoadtyz .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#tyxwoadtyz .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#tyxwoadtyz .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#tyxwoadtyz .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#tyxwoadtyz .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tyxwoadtyz .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#tyxwoadtyz .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#tyxwoadtyz .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#tyxwoadtyz .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#tyxwoadtyz .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#tyxwoadtyz .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#tyxwoadtyz .gt_spanner_row {
  border-bottom-style: hidden;
}

#tyxwoadtyz .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#tyxwoadtyz .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#tyxwoadtyz .gt_from_md > :first-child {
  margin-top: 0;
}

#tyxwoadtyz .gt_from_md > :last-child {
  margin-bottom: 0;
}

#tyxwoadtyz .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#tyxwoadtyz .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#tyxwoadtyz .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#tyxwoadtyz .gt_row_group_first td {
  border-top-width: 2px;
}

#tyxwoadtyz .gt_row_group_first th {
  border-top-width: 2px;
}

#tyxwoadtyz .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#tyxwoadtyz .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#tyxwoadtyz .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#tyxwoadtyz .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tyxwoadtyz .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#tyxwoadtyz .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#tyxwoadtyz .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#tyxwoadtyz .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#tyxwoadtyz .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tyxwoadtyz .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#tyxwoadtyz .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#tyxwoadtyz .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#tyxwoadtyz .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#tyxwoadtyz .gt_left {
  text-align: left;
}

#tyxwoadtyz .gt_center {
  text-align: center;
}

#tyxwoadtyz .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#tyxwoadtyz .gt_font_normal {
  font-weight: normal;
}

#tyxwoadtyz .gt_font_bold {
  font-weight: bold;
}

#tyxwoadtyz .gt_font_italic {
  font-style: italic;
}

#tyxwoadtyz .gt_super {
  font-size: 65%;
}

#tyxwoadtyz .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#tyxwoadtyz .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#tyxwoadtyz .gt_indent_1 {
  text-indent: 5px;
}

#tyxwoadtyz .gt_indent_2 {
  text-indent: 10px;
}

#tyxwoadtyz .gt_indent_3 {
  text-indent: 15px;
}

#tyxwoadtyz .gt_indent_4 {
  text-indent: 20px;
}

#tyxwoadtyz .gt_indent_5 {
  text-indent: 25px;
}

#tyxwoadtyz .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#tyxwoadtyz div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<colgroup>
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="gt_heading header">
<th colspan="7" class="gt_heading gt_title gt_font_normal"><strong>Functional Richness (FR)</strong></th>
</tr>
<tr class="gt_heading even">
<th colspan="7" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border">FR Palms and FR Mammals model coefficients</th>
</tr>
<tr class="gt_col_headings gt_spanner_row header">
<th rowspan="2" id="Predictor" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Predictors</th>
<th colspan="3" id="**FR Palms**" class="gt_center gt_columns_top_border gt_column_spanner_outer" data-quarto-table-cell-role="th" scope="colgroup"><div class="gt_column_spanner">
<strong>FR Palms</strong>
</div></th>
<th colspan="3" id="**FR Mammals**" class="gt_center gt_columns_top_border gt_column_spanner_outer" data-quarto-table-cell-role="th" scope="colgroup"><div class="gt_column_spanner">
<strong>FR Mammals</strong>
</div></th>
</tr>
<tr class="gt_col_headings even">
<th id="Palms_Estimate" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Estimates</th>
<th id="Palms_SE" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Std. Error</th>
<th id="Palms_p" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p</th>
<th id="Mammals_Estimate" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Estimates</th>
<th id="Mammals_SE" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Std. Error</th>
<th id="Mammals_p" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">p</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">(Intercept)</td>
<td class="gt_row gt_left" headers="Palms_Estimate">8.28e-03</td>
<td class="gt_row gt_left" headers="Palms_SE">8.58e-03</td>
<td class="gt_row gt_right" headers="Palms_p">0.335</td>
<td class="gt_row gt_left" headers="Mammals_Estimate">1.35e-01</td>
<td class="gt_row gt_left" headers="Mammals_SE">1.52e-03</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">&lt;0.001</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">Mean Annual Temperature (MAT)</td>
<td class="gt_row gt_left" headers="Palms_Estimate">4.07e-04</td>
<td class="gt_row gt_left" headers="Palms_SE">1.36e-04</td>
<td class="gt_row gt_right" headers="Palms_p" style="font-weight: bold">0.003</td>
<td class="gt_row gt_left" headers="Mammals_Estimate">4.42e-04</td>
<td class="gt_row gt_left" headers="Mammals_SE">4.48e-05</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">&lt;0.001</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">Total Annual Precipitation (TAP)</td>
<td class="gt_row gt_left" headers="Palms_Estimate">1.75e-05</td>
<td class="gt_row gt_left" headers="Palms_SE">1.47e-05</td>
<td class="gt_row gt_right" headers="Palms_p">0.233</td>
<td class="gt_row gt_left" headers="Mammals_Estimate">6.42e-06</td>
<td class="gt_row gt_left" headers="Mammals_SE">4.95e-06</td>
<td class="gt_row gt_left" headers="Mammals_p">0.153</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">Temperature Seasonality (TS)</td>
<td class="gt_row gt_left" headers="Palms_Estimate">-4.61e-05</td>
<td class="gt_row gt_left" headers="Palms_SE">1.72e-04</td>
<td class="gt_row gt_right" headers="Palms_p">0.789</td>
<td class="gt_row gt_left" headers="Mammals_Estimate">3.88e-05</td>
<td class="gt_row gt_left" headers="Mammals_SE">8.13e-05</td>
<td class="gt_row gt_left" headers="Mammals_p">0.633</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">Precipitation Seasonality (PS)</td>
<td class="gt_row gt_left" headers="Palms_Estimate">1.68e-04</td>
<td class="gt_row gt_left" headers="Palms_SE">2.48e-04</td>
<td class="gt_row gt_right" headers="Palms_p">0.499</td>
<td class="gt_row gt_left" headers="Mammals_Estimate">2.06e-04</td>
<td class="gt_row gt_left" headers="Mammals_SE">1.35e-04</td>
<td class="gt_row gt_left" headers="Mammals_p">0.128</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">MAT × IG-1</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.073</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">0.011</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">MAT × IG-2</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.129</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">0.010</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">MAT × IG-3</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p" style="font-weight: bold">0.014</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.331</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">MAT × IG-4</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.073</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">0.001</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">MAT × IG-5</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.067</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">0.010</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">MAT × IG-6</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.207</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">0.002</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">MAT × IG-7</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.073</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p" style="font-weight: bold">0.003</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">TAP × IG-1</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.331</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.322</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">TAP × IG-2</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.292</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.541</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">TAP × IG-3</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.392</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.552</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">TAP × IG-4</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.331</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.449</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">TAP × IG-5</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.637</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.541</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">TAP × IG-6</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p" style="font-weight: bold">0.007</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.950</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">TAP × IG-7</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.331</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.436</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">TS × IG-1</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.864</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.654</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">TS × IG-2</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.863</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.849</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">TS × IG-3</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.868</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.748</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">TS × IG-4</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.864</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.678</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">TS × IG-5</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.890</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.849</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">TS × IG-6</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.352</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.920</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">TS × IG-7</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.864</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.611</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">PS × IG-1</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.705</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.225</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">PS × IG-2</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.199</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.568</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">PS × IG-3</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.239</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.748</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">PS × IG-4</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.705</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.314</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">PS × IG-5</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p" style="font-weight: bold">0.007</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.568</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">PS × IG-6</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p" style="font-weight: bold">0.001</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.549</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">PS × IG-7</td>
<td class="gt_row gt_left" headers="Palms_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Palms_SE"><br>
</td>
<td class="gt_row gt_right" headers="Palms_p">0.705</td>
<td class="gt_row gt_left" headers="Mammals_Estimate"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_SE"><br>
</td>
<td class="gt_row gt_left" headers="Mammals_p">0.361</td>
</tr>
</tbody><tfoot class="gt_sourcenotes">
<tr class="odd">
<td colspan="7" class="gt_sourcenote">Observations = 9,009 (both models) | R² (Palms) = 0.012 | R² (Mammals) = -0.002</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
</div>
</section>
<section id="measuring-fta" class="level3" data-number="B.3.3">
<h3 data-number="B.3.3" class="anchored" data-anchor-id="measuring-fta"><span class="header-section-number">B.3.3</span> Measuring FTA</h3>
<p>Functional trophic asymmetry was calculated as the absolute difference between the normalized palm and mammal guild proportions, capturing the degree to which interaction guild composition differs between resources and consumers within the same assemblage. This resulted on 49 individual metrics (7 x 7 ) of FTA per gridcell, each representing a unique interaction guild combination.</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 7
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[\mathrm{FTA}_{g_ig_j} = \left|Fr^{P}_{g_i} - Fr^{M}_{g_j}\right|\]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(\mathrm{FTA}_{g_i g_j}\)</span>) is the functional trophic asymmetry between palms and mammals for interaction guilds combinations between palms (<span class="math inline">\(g_i\)</span>) and mammals (<span class="math inline">\(g_j\)</span>); (<span class="math inline">\(Fr^{P}_{g_i}\)</span>) is the relative proportion of palm species assigned to guild (<span class="math inline">\(g_i\)</span>); and (<span class="math inline">\(Fr^{M}_{g_j}\)</span>) is the relative proportion of mammal species assigned to guild (<span class="math inline">\(g_j\)</span>).</p>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-2134524700.png" class="img-fluid figure-img"></p>
<figcaption>Figure S4: Ridgeline densities of Functional Trophic Asymmetry (FTA) for each realized mammal–palm “interaction guild” (mXpY) across Neotropical grid cells. Each label on the y-axis—e.g.&nbsp;m3p2—denotes the pairing of mammal guild 3 (small-bodied, low-frugivory, predominantly nocturnal species: mean mass ≈ 4.75 kg; 21.7 % fruit diet; 84 % nocturnal) with palm guild 2 (tall, small-fruited, erect palms: 25 species; mean height ≈ 3.05 m; fruit length ≈ 0.8 cm; 100 % erect). The x-axis shows FTA values (0 = symmetrical trait distributions, 1 = maximal mismatch), and each ridge plots the distribution of FTA across all grid cells where that guild pair co-occurs. Guilds combining large, low-frugivory mammals (e.g.&nbsp;m6) with short, large-fruited palms (p6) (see table 3 for trait distribution reference) exhibit the highest asymmetry (densities shifted right), whereas guilds of small, highly frugivorous mammals paired with tall, small-fruited palms (e.g.&nbsp;m1p2) cluster near zero.</figcaption>
</figure>
</div>
<p>We summed FTA values across all interaction guild combinations to obtain an aggregate metric of FTA per grid cell <span class="math inline">\(FTA_{sum}\)</span>.</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 8
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[\mathrm{FTA}_{\mathrm{sum}} = \sum_{g=1}^{G}  \mathrm{FTA_g} \]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(\mathrm{FTA}_{sum}\)</span>) is the aggregate functional trophic asymmetry across all interaction guilds; (<span class="math inline">\(G\)</span>) is the total number of palm-mammal interaction guilds combinations inferred by the stochastic block model (49); (<span class="math inline">\(\mathrm{FTA}\)</span>) is the functional trophic asymmetry between palms and mammals guilds.</p>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-2272296321.png" class="img-fluid figure-img"></p>
<figcaption>**Figure X. Distribution of summed Functional Trophic Asymmetry (FTA(_{\mathrm{sum}})) across Neotropical grid cells.** The histogram shows the frequency distribution of (\mathrm{FTA}_{\mathrm{sum}}), which aggregates absolute trait asymmetries across all interacting mammal–palm guild pairs within each grid cell. Higher values indicate greater overall mismatch in functional traits between consumer and resource assemblages, whereas lower values reflect more symmetric guild composition.</figcaption>
</figure>
</div>
<p>Because <span class="math inline">\(FTA_{sum}\)</span> can be sensitive to asymmetries arising in weakly connected parts of the network, weighting functional trophic asymmetry by inter-guild interaction strength accounts for the fact that not all guild mismatches are equally relevant to network organization. The adjusted metric therefore prioritizes asymmetry in strongly interacting guild pairs, providing a more ecologically meaningful measure of trophic imbalance.</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Equation 9
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math display">\[\mathrm{FTA}_{sum_{adj}} = \sum_{g=1}^{G} FTA \times I_{g_ig_j}\]</span></p>
<p><strong>where:</strong> (<span class="math inline">\(I_{g_i g_j}\)</span>) is the strenght of interaction between palm-mammal guilds.</p>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-3422516409.png" class="img-fluid figure-img"></p>
<figcaption>**Figure X. Distribution of adjusted summed Functional Trophic Asymmetry (FTA(_{\mathrm{sum,adj}})) across Neotropical grid cells.** The histogram depicts the frequency distribution of the interaction-weighted FTA metric, in which trait asymmetries among mammal–palm guild pairs are scaled by the strength of their inferred interactions. This adjustment down-weights rare or weakly connected guild combinations, emphasizing asymmetry arising from ecologically dominant interactions and providing a more conservative measure of functional mismatch at the assemblage level.</figcaption>
</figure>
</div>
<section id="evaluating-the-relationship-between-fta-and-interaction-strength" class="level4" data-number="B.3.3.1">
<h4 data-number="B.3.3.1" class="anchored" data-anchor-id="evaluating-the-relationship-between-fta-and-interaction-strength"><span class="header-section-number">B.3.3.1</span> Evaluating the relationship between FTA and interaction strength</h4>
<p>To ensure that the interaction‐weighted adjustment did not introduce systematic bias, we explicitly tested whether the <span class="math inline">\(FTA_{sum_{adj}}\)</span>was correlated with interaction strength. <em>T</em>his assessment verifies that the adjustment does not trivially inflate or deflate asymmetry values as a direct function of interaction intensity, but instead preserves FTA as an independent measure of trait mismatch.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-924046659.png" class="img-fluid figure-img"></p>
<figcaption>Figure S7: Relationship between interaction strength and Functional Trophic Asymmetry (FTA) for 43/49 grid‐cell–interaction guild observations. Points show the median FTA values plotted against interaction strength, with the blue line and grey band representing the fitted linear regression (±95 % CI). The model intercept significantly different from 0, 0.19 (95 % CI 0.12–0.25, p &lt; 0.001), but the slope for interaction strength is small and non‐significant (β = 0.18, 95 % CI –0.27 to 0.62, p = 0.426).</figcaption>
</figure>
</div>
<div class="callout callout-style-simple callout-none no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p>####Table B3: Simple linear regression of adjusted Functional Trophic Asymmetry (FTA) on interaction strength.</p>
<p>The intercept of 0.19 (95 % CI: 0.12–0.25, p &lt; 0.001) represents the baseline FTA when interaction strength is zero, while the slope coefficient for interaction strength is 0.18 (95 % CI: –0.27–0.62, p = 0.426), indicating that variation in interaction strength does not significantly predict changes in FTA.</p>
<div class="cell">
<div class="cell-output-display">
<div id="asgjodzwmg" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#asgjodzwmg table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#asgjodzwmg thead, #asgjodzwmg tbody, #asgjodzwmg tfoot, #asgjodzwmg tr, #asgjodzwmg td, #asgjodzwmg th {
  border-style: none;
}

#asgjodzwmg p {
  margin: 0;
  padding: 0;
}

#asgjodzwmg .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 3px;
  border-top-color: #D3D3D3;
  border-right-style: solid;
  border-right-width: 3px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 3px;
  border-bottom-color: #D3D3D3;
  border-left-style: solid;
  border-left-width: 3px;
  border-left-color: #D3D3D3;
}

#asgjodzwmg .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#asgjodzwmg .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#asgjodzwmg .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#asgjodzwmg .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#asgjodzwmg .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#asgjodzwmg .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#asgjodzwmg .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#asgjodzwmg .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#asgjodzwmg .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#asgjodzwmg .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#asgjodzwmg .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#asgjodzwmg .gt_spanner_row {
  border-bottom-style: hidden;
}

#asgjodzwmg .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#asgjodzwmg .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#asgjodzwmg .gt_from_md > :first-child {
  margin-top: 0;
}

#asgjodzwmg .gt_from_md > :last-child {
  margin-bottom: 0;
}

#asgjodzwmg .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#asgjodzwmg .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#asgjodzwmg .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#asgjodzwmg .gt_row_group_first td {
  border-top-width: 2px;
}

#asgjodzwmg .gt_row_group_first th {
  border-top-width: 2px;
}

#asgjodzwmg .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#asgjodzwmg .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#asgjodzwmg .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#asgjodzwmg .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#asgjodzwmg .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#asgjodzwmg .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#asgjodzwmg .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#asgjodzwmg .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#asgjodzwmg .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#asgjodzwmg .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#asgjodzwmg .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#asgjodzwmg .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#asgjodzwmg .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#asgjodzwmg .gt_left {
  text-align: left;
}

#asgjodzwmg .gt_center {
  text-align: center;
}

#asgjodzwmg .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#asgjodzwmg .gt_font_normal {
  font-weight: normal;
}

#asgjodzwmg .gt_font_bold {
  font-weight: bold;
}

#asgjodzwmg .gt_font_italic {
  font-style: italic;
}

#asgjodzwmg .gt_super {
  font-size: 65%;
}

#asgjodzwmg .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#asgjodzwmg .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#asgjodzwmg .gt_indent_1 {
  text-indent: 5px;
}

#asgjodzwmg .gt_indent_2 {
  text-indent: 10px;
}

#asgjodzwmg .gt_indent_3 {
  text-indent: 15px;
}

#asgjodzwmg .gt_indent_4 {
  text-indent: 20px;
}

#asgjodzwmg .gt_indent_5 {
  text-indent: 25px;
}

#asgjodzwmg .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#asgjodzwmg div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_heading header">
<th colspan="4" class="gt_heading gt_title gt_font_normal"><strong>Model Results</strong></th>
</tr>
<tr class="gt_heading even">
<th colspan="4" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border">Coefficient estimates with 95% confidence intervals</th>
</tr>
<tr class="gt_col_headings header">
<th id="Predictor" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Predictors</th>
<th id="Estimate" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Estimates</th>
<th id="CI" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">CI</th>
<th id="p" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">p</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">(Intercept)</td>
<td class="gt_row gt_right" headers="Estimate">0.19</td>
<td class="gt_row gt_left" headers="CI">0.12 – 0.25</td>
<td class="gt_row gt_left" headers="p" style="font-weight: bold">&lt;0.001</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">Interaction strength</td>
<td class="gt_row gt_right" headers="Estimate">0.18</td>
<td class="gt_row gt_left" headers="CI">-0.27 – 0.62</td>
<td class="gt_row gt_left" headers="p">0.426</td>
</tr>
</tbody><tfoot class="gt_sourcenotes">
<tr class="odd">
<td colspan="4" class="gt_sourcenote">Observations = 43 | R² / R² adjusted = 0.016 / -0.008</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="does-climate-determine-fta" class="level3" data-number="B.3.4">
<h3 data-number="B.3.4" class="anchored" data-anchor-id="does-climate-determine-fta"><span class="header-section-number">B.3.4</span> Does climate determine FTA?</h3>
<p>We used generalized additive models (mgcv) to relate the response to climatic predictors—mean annual temperature (Temp), precipitation seasonality (PS), temperature seasonality (TS), and total annual precipitation (Prec)—and to interaction strength (int_str). Each climate variable was included as a univariate smooth s(⋅) (thin-plate regression spline; k=9k=9k=9), and potential modulation by interaction strength was modeled using smooth interaction terms s(climate,int_str)(k=27). A separate smooth s(int_str) captured nonlinear effects of interaction strength. Smoothing parameters were selected by generalized cross-validation (GCV) using the <em>magic</em> optimizer. Model adequacy was evaluated using mgcv basis-dimension checks (k-index), effective degrees of freedom, and residual diagnostics.</p>
<p>Checks on the basis dimensions show no evidence that any smooth term is underfit: the climatic predictors (temperature, precipitation, and their seasonalities) have effective degrees of freedom close to one and k-indices near one with high p-values, indicating that their effects are approximately linear and adequately captured by the chosen smooths. Interaction terms between climate variables and interaction strength show very low complexity. In contrast, the smooth term for interaction strength alone is highly flexible and strongly supported, indicating a pronounced nonlinear relationship. Convergence diagnostics indicate a stable and well-conditioned model.</p>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Table B4. Basis-dimension diagnostics for generalized additive model smooth terms.
</div>
</div>
<div class="callout-body-container callout-body">
<p>The table reports the basis dimension ((k′)), effective degrees of freedom (EDF), k-index, and associated p-values for each smooth term.</p>
<div class="cell">
<div class="cell-output-display">
<div id="ngsdrglvhk" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ngsdrglvhk table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ngsdrglvhk thead, #ngsdrglvhk tbody, #ngsdrglvhk tfoot, #ngsdrglvhk tr, #ngsdrglvhk td, #ngsdrglvhk th {
  border-style: none;
}

#ngsdrglvhk p {
  margin: 0;
  padding: 0;
}

#ngsdrglvhk .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 3px;
  border-top-color: #D3D3D3;
  border-right-style: solid;
  border-right-width: 3px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 3px;
  border-bottom-color: #D3D3D3;
  border-left-style: solid;
  border-left-width: 3px;
  border-left-color: #D3D3D3;
}

#ngsdrglvhk .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ngsdrglvhk .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ngsdrglvhk .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ngsdrglvhk .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ngsdrglvhk .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ngsdrglvhk .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ngsdrglvhk .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ngsdrglvhk .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ngsdrglvhk .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ngsdrglvhk .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ngsdrglvhk .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ngsdrglvhk .gt_spanner_row {
  border-bottom-style: hidden;
}

#ngsdrglvhk .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ngsdrglvhk .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ngsdrglvhk .gt_from_md > :first-child {
  margin-top: 0;
}

#ngsdrglvhk .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ngsdrglvhk .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ngsdrglvhk .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ngsdrglvhk .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ngsdrglvhk .gt_row_group_first td {
  border-top-width: 2px;
}

#ngsdrglvhk .gt_row_group_first th {
  border-top-width: 2px;
}

#ngsdrglvhk .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ngsdrglvhk .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ngsdrglvhk .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ngsdrglvhk .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ngsdrglvhk .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ngsdrglvhk .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ngsdrglvhk .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ngsdrglvhk .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ngsdrglvhk .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ngsdrglvhk .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ngsdrglvhk .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ngsdrglvhk .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ngsdrglvhk .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ngsdrglvhk .gt_left {
  text-align: left;
}

#ngsdrglvhk .gt_center {
  text-align: center;
}

#ngsdrglvhk .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ngsdrglvhk .gt_font_normal {
  font-weight: normal;
}

#ngsdrglvhk .gt_font_bold {
  font-weight: bold;
}

#ngsdrglvhk .gt_font_italic {
  font-style: italic;
}

#ngsdrglvhk .gt_super {
  font-size: 65%;
}

#ngsdrglvhk .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ngsdrglvhk .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ngsdrglvhk .gt_indent_1 {
  text-indent: 5px;
}

#ngsdrglvhk .gt_indent_2 {
  text-indent: 10px;
}

#ngsdrglvhk .gt_indent_3 {
  text-indent: 15px;
}

#ngsdrglvhk .gt_indent_4 {
  text-indent: 20px;
}

#ngsdrglvhk .gt_indent_5 {
  text-indent: 25px;
}

#ngsdrglvhk .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#ngsdrglvhk div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_heading header">
<th colspan="6" class="gt_heading gt_title gt_font_normal"><strong>GAM basis-dimension diagnostics</strong></th>
</tr>
<tr class="gt_heading even">
<th colspan="6" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border">Results of k-index and effective degrees of freedom checks</th>
</tr>
<tr class="gt_col_headings header">
<th id="Smooth" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Smooth term</th>
<th id="k_prime" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">k′</th>
<th id="edf" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">EDF</th>
<th id="k_index" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">k-index</th>
<th id="p_value" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">p</th>
<th id="sig" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col"></th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Smooth">s(Temp)</td>
<td class="gt_row gt_center" headers="k_prime">9.00</td>
<td class="gt_row gt_center" headers="edf">1.00</td>
<td class="gt_row gt_center" headers="k_index">1.02</td>
<td class="gt_row gt_center" headers="p_value">0.80</td>
<td class="gt_row gt_center" headers="sig"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left gt_striped" headers="Smooth">s(Temp, int_str)</td>
<td class="gt_row gt_center gt_striped" headers="k_prime">27.00</td>
<td class="gt_row gt_center gt_striped" headers="edf">0.01</td>
<td class="gt_row gt_center gt_striped" headers="k_index">1.02</td>
<td class="gt_row gt_center gt_striped" headers="p_value">0.96</td>
<td class="gt_row gt_center gt_striped" headers="sig"></td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Smooth">s(PS)</td>
<td class="gt_row gt_center" headers="k_prime">9.00</td>
<td class="gt_row gt_center" headers="edf">1.00</td>
<td class="gt_row gt_center" headers="k_index">1.03</td>
<td class="gt_row gt_center" headers="p_value">0.97</td>
<td class="gt_row gt_center" headers="sig"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left gt_striped" headers="Smooth">s(PS, int_str)</td>
<td class="gt_row gt_center gt_striped" headers="k_prime">27.00</td>
<td class="gt_row gt_center gt_striped" headers="edf">1.77</td>
<td class="gt_row gt_center gt_striped" headers="k_index">1.02</td>
<td class="gt_row gt_center gt_striped" headers="p_value">0.94</td>
<td class="gt_row gt_center gt_striped" headers="sig"></td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Smooth">s(TS)</td>
<td class="gt_row gt_center" headers="k_prime">9.00</td>
<td class="gt_row gt_center" headers="edf">1.00</td>
<td class="gt_row gt_center" headers="k_index">1.02</td>
<td class="gt_row gt_center" headers="p_value">0.96</td>
<td class="gt_row gt_center" headers="sig"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left gt_striped" headers="Smooth">s(TS, int_str)</td>
<td class="gt_row gt_center gt_striped" headers="k_prime">27.00</td>
<td class="gt_row gt_center gt_striped" headers="edf">0.00</td>
<td class="gt_row gt_center gt_striped" headers="k_index">1.02</td>
<td class="gt_row gt_center gt_striped" headers="p_value">0.87</td>
<td class="gt_row gt_center gt_striped" headers="sig"></td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Smooth">s(Prec)</td>
<td class="gt_row gt_center" headers="k_prime">9.00</td>
<td class="gt_row gt_center" headers="edf">1.00</td>
<td class="gt_row gt_center" headers="k_index">1.01</td>
<td class="gt_row gt_center" headers="p_value">0.73</td>
<td class="gt_row gt_center" headers="sig"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left gt_striped" headers="Smooth">s(Prec, int_str)</td>
<td class="gt_row gt_center gt_striped" headers="k_prime">27.00</td>
<td class="gt_row gt_center gt_striped" headers="edf">1.74</td>
<td class="gt_row gt_center gt_striped" headers="k_index">1.01</td>
<td class="gt_row gt_center gt_striped" headers="p_value">0.68</td>
<td class="gt_row gt_center gt_striped" headers="sig"></td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Smooth" style="font-weight: bold">s(int_str)</td>
<td class="gt_row gt_center" headers="k_prime" style="font-weight: bold">9.00</td>
<td class="gt_row gt_center" headers="edf" style="font-weight: bold">9.00</td>
<td class="gt_row gt_center" headers="k_index" style="font-weight: bold">0.62</td>
<td class="gt_row gt_center" headers="p_value" style="font-weight: bold">&lt; 2e-16</td>
<td class="gt_row gt_center" headers="sig" style="font-weight: bold">***</td>
</tr>
</tbody><tfoot class="gt_sourcenotes">
<tr class="odd">
<td colspan="6" class="gt_sourcenote">Low k-index values (&lt; 1) with significant p-values indicate potential undersmoothing. Only the smooth for interaction strength shows evidence of requiring high flexibility.</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-3768095388.png" class="img-fluid figure-img"></p>
<figcaption>Figure S8. Estimated smooth relationships from a GAM of Functional Trophic Asymmetry (FTA) against four climatic variables and interaction strength. Across all panels, mean annual temperature (Temp) shows a gentle, positive association with FTA, while total precipitation and temperature seasonality exert negligible or slightly negative effects. Precipitation (Prec) seasonality likewise produces only a weak upward trend at high values. Importantly, none of the tensor‐product interaction terms between interaction strength and any climate variable deviate meaningfully from zero, indicating that interaction strength (int_str) does not modify how climate influences trait asymmetry. The standalone smooth for interaction strength itself exhibits undulating but statistically uncertain variation, suggesting no clear direct effect. Shaded bands around each curve denote approximately 95 % confidence intervals, and tick marks along the axes mark the range of observed data.</figcaption>
</figure>
</div>
</section>
<section id="does-fta-influence-network-structure" class="level3" data-number="B.3.5">
<h3 data-number="B.3.5" class="anchored" data-anchor-id="does-fta-influence-network-structure"><span class="header-section-number">B.3.5</span> Does FTA influence network structure?</h3>
<p>We examine the variation in network structure with network specialization and the spectral radius. We computed a metric of network-level specialization (H2’) that is independent of network size and marginal totals. The metric is computed from the interaction matrix by comparing the deviation of interaction frequencies from a null expectation based on species’ total interaction frequencies, and is then normalized to range between 0 (no specialization, complete generalization) and 1 (maximum specialization). In addition, we used the spectral radius as a complementary metric of network structure. The spectral radius is computed as the largest eigenvalue of the (weighted or binary) interaction matrix. This metric captures the dominant mode of interaction structure in the network, reflecting how strongly interactions are concentrated along a few highly connected species or pathways. A larger spectral radius indicates a more centralized interaction structure dominated by generalists, whereas lower values suggest interactions that are more evenly distributed creating diffuse networks.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-3266454264.png" class="img-fluid figure-img"></p>
<figcaption>Figure X. Relationship between network specialization and structural concentration. Scatterplot showing grid-cell values of network specialization (H2′) plotted against the log-transformed spectral radius of the interaction matrix.</figcaption>
</figure>
</div>
<p>We control for differences in sampling intensity and network size by standardizing interaction data to a common level of observation. Specifically, we used <code>rarefy</code> function from <code>cassandRa</code> to repeatedly subsample interactions from the original bipartite gridcell level network to a fixed number (100) of interactions and recalculates network structural metrics (i.e network specialization, structural radius) on each subsample. We used the median of this subsampled distribution as the gridcell metric of network structure. This procedure reduces biases that arise because larger or more densely sampled assemblages tend to exhibit higher connectance.</p>
<p><img src="images/clipboard-321485642.png" class="img-fluid"></p>
<section id="evaluating-the-relationship-between-fta-and-h2" class="level4" data-number="B.3.5.1">
<h4 data-number="B.3.5.1" class="anchored" data-anchor-id="evaluating-the-relationship-between-fta-and-h2"><span class="header-section-number">B.3.5.1</span> Evaluating the relationship between FTA and H2’</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-3970149461.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure S9.Functional trophic asymmetry predicts network specialization.</strong> Scatterplot shows the relationship between adjusted functional trophic asymmetry and network specialization across grid cell and guild observations in the Neotropics. Each point represents a local assemblage, illustrating how increasing trophic asymmetry is associated with higher levels of interaction specialization</figcaption>
</figure>
</div>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Table B5: Simple linear model relating network specialization (H2′) to total functional trophic asymmetry (FTAsum).
</div>
</div>
<div class="callout-body-container callout-body">
<p>The model intercept is –0.03 (95 % CI: –0.13–0.06, p = 0.489), and FTAsum has a positive estimate of 0.05 (95 % CI: 0.02–0.07, p &lt; 0.001), suggesting a small but statistically significant increase in specialization with greater trait mismatch.</p>
<div class="cell">
<div class="cell-output-display">
<div id="dwvgndrohd" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#dwvgndrohd table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#dwvgndrohd thead, #dwvgndrohd tbody, #dwvgndrohd tfoot, #dwvgndrohd tr, #dwvgndrohd td, #dwvgndrohd th {
  border-style: none;
}

#dwvgndrohd p {
  margin: 0;
  padding: 0;
}

#dwvgndrohd .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 3px;
  border-top-color: #D3D3D3;
  border-right-style: solid;
  border-right-width: 3px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 3px;
  border-bottom-color: #D3D3D3;
  border-left-style: solid;
  border-left-width: 3px;
  border-left-color: #D3D3D3;
}

#dwvgndrohd .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#dwvgndrohd .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#dwvgndrohd .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#dwvgndrohd .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dwvgndrohd .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dwvgndrohd .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dwvgndrohd .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#dwvgndrohd .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#dwvgndrohd .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#dwvgndrohd .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#dwvgndrohd .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#dwvgndrohd .gt_spanner_row {
  border-bottom-style: hidden;
}

#dwvgndrohd .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#dwvgndrohd .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#dwvgndrohd .gt_from_md > :first-child {
  margin-top: 0;
}

#dwvgndrohd .gt_from_md > :last-child {
  margin-bottom: 0;
}

#dwvgndrohd .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#dwvgndrohd .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#dwvgndrohd .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#dwvgndrohd .gt_row_group_first td {
  border-top-width: 2px;
}

#dwvgndrohd .gt_row_group_first th {
  border-top-width: 2px;
}

#dwvgndrohd .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dwvgndrohd .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#dwvgndrohd .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#dwvgndrohd .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dwvgndrohd .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dwvgndrohd .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#dwvgndrohd .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#dwvgndrohd .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#dwvgndrohd .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dwvgndrohd .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dwvgndrohd .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dwvgndrohd .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dwvgndrohd .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dwvgndrohd .gt_left {
  text-align: left;
}

#dwvgndrohd .gt_center {
  text-align: center;
}

#dwvgndrohd .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#dwvgndrohd .gt_font_normal {
  font-weight: normal;
}

#dwvgndrohd .gt_font_bold {
  font-weight: bold;
}

#dwvgndrohd .gt_font_italic {
  font-style: italic;
}

#dwvgndrohd .gt_super {
  font-size: 65%;
}

#dwvgndrohd .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#dwvgndrohd .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#dwvgndrohd .gt_indent_1 {
  text-indent: 5px;
}

#dwvgndrohd .gt_indent_2 {
  text-indent: 10px;
}

#dwvgndrohd .gt_indent_3 {
  text-indent: 15px;
}

#dwvgndrohd .gt_indent_4 {
  text-indent: 20px;
}

#dwvgndrohd .gt_indent_5 {
  text-indent: 25px;
}

#dwvgndrohd .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#dwvgndrohd div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_heading header">
<th colspan="4" class="gt_heading gt_title gt_font_normal"><strong>Network specialization</strong></th>
</tr>
<tr class="gt_heading even">
<th colspan="4" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border">Coefficient estimates with 95% confidence intervals</th>
</tr>
<tr class="gt_col_headings header">
<th id="Predictor" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Predictors</th>
<th id="Estimate" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Estimates</th>
<th id="CI" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">CI</th>
<th id="p" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">p</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Predictor">(Intercept)</td>
<td class="gt_row gt_right" headers="Estimate">-0.03</td>
<td class="gt_row gt_left" headers="CI">-0.13 – 0.06</td>
<td class="gt_row gt_left" headers="p">0.489</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Predictor">FTAsum</td>
<td class="gt_row gt_right" headers="Estimate">0.05</td>
<td class="gt_row gt_left" headers="CI">0.02 – 0.07</td>
<td class="gt_row gt_left" headers="p" style="font-weight: bold">&lt;0.001</td>
</tr>
</tbody><tfoot class="gt_sourcenotes">
<tr class="odd">
<td colspan="4" class="gt_sourcenote">Observations = 1,242 | R² / R² adjusted = 0.012 / 0.012</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
</div>
</section>
<section id="evaluating-the-relationship-between-fta-and-the-spectral-radius" class="level4" data-number="B.3.5.2">
<h4 data-number="B.3.5.2" class="anchored" data-anchor-id="evaluating-the-relationship-between-fta-and-the-spectral-radius"><span class="header-section-number">B.3.5.2</span> Evaluating the relationship between FTA and the spectral radius</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clipboard-1692998543.png" class="img-fluid figure-img"></p>
<figcaption><strong>Figure X. Functional trophic asymmetry predicts network spectral radius.</strong> Scatterplot shows the relationship between adjusted functional trophic asymmetry and the logarithm of the spectral radius of interaction networks after rarefaction to control for sampling effects. Each point represents a rarefied network realization, with the fitted regression line and confidence interval illustrating a strong positive association.</figcaption>
</figure>
</div>
<div class="callout callout-style-simple callout-none no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Table B6: Effect of functional trophic asymmetry on network spectral radius.
</div>
</div>
<div class="callout-body-container callout-body">
<p>The table show the results of a simple linear regression showing the relationship between functional trophic asymmetry and spectral radius. Estimates indicate a strong positive association, with higher asymmetry corresponding to networks that are more ordered, with lower-dimensional interaction structure.</p>
<div class="cell">
<div class="cell-output-display">
<div id="yylkbxpqay" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#yylkbxpqay table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#yylkbxpqay thead, #yylkbxpqay tbody, #yylkbxpqay tfoot, #yylkbxpqay tr, #yylkbxpqay td, #yylkbxpqay th {
  border-style: none;
}

#yylkbxpqay p {
  margin: 0;
  padding: 0;
}

#yylkbxpqay .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 3px;
  border-top-color: #D3D3D3;
  border-right-style: solid;
  border-right-width: 3px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 3px;
  border-bottom-color: #D3D3D3;
  border-left-style: solid;
  border-left-width: 3px;
  border-left-color: #D3D3D3;
}

#yylkbxpqay .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#yylkbxpqay .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#yylkbxpqay .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#yylkbxpqay .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#yylkbxpqay .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yylkbxpqay .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#yylkbxpqay .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#yylkbxpqay .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#yylkbxpqay .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#yylkbxpqay .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#yylkbxpqay .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#yylkbxpqay .gt_spanner_row {
  border-bottom-style: hidden;
}

#yylkbxpqay .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#yylkbxpqay .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#yylkbxpqay .gt_from_md > :first-child {
  margin-top: 0;
}

#yylkbxpqay .gt_from_md > :last-child {
  margin-bottom: 0;
}

#yylkbxpqay .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#yylkbxpqay .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#yylkbxpqay .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#yylkbxpqay .gt_row_group_first td {
  border-top-width: 2px;
}

#yylkbxpqay .gt_row_group_first th {
  border-top-width: 2px;
}

#yylkbxpqay .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#yylkbxpqay .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#yylkbxpqay .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#yylkbxpqay .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yylkbxpqay .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#yylkbxpqay .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#yylkbxpqay .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#yylkbxpqay .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#yylkbxpqay .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yylkbxpqay .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#yylkbxpqay .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#yylkbxpqay .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#yylkbxpqay .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#yylkbxpqay .gt_left {
  text-align: left;
}

#yylkbxpqay .gt_center {
  text-align: center;
}

#yylkbxpqay .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#yylkbxpqay .gt_font_normal {
  font-weight: normal;
}

#yylkbxpqay .gt_font_bold {
  font-weight: bold;
}

#yylkbxpqay .gt_font_italic {
  font-style: italic;
}

#yylkbxpqay .gt_super {
  font-size: 65%;
}

#yylkbxpqay .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#yylkbxpqay .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#yylkbxpqay .gt_indent_1 {
  text-indent: 5px;
}

#yylkbxpqay .gt_indent_2 {
  text-indent: 10px;
}

#yylkbxpqay .gt_indent_3 {
  text-indent: 15px;
}

#yylkbxpqay .gt_indent_4 {
  text-indent: 20px;
}

#yylkbxpqay .gt_indent_5 {
  text-indent: 25px;
}

#yylkbxpqay .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#yylkbxpqay div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="gt_heading header">
<th colspan="5" class="gt_heading gt_title gt_font_normal"><strong>Spectral radius</strong></th>
</tr>
<tr class="gt_heading even">
<th colspan="5" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border">Coefficient estimates with 95% confidence intervals</th>
</tr>
<tr class="gt_col_headings header">
<th id="Term" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Term</th>
<th id="Estimate" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Estimate</th>
<th id="Std.-Error" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Std. Error</th>
<th id="t-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">t value</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">p value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Term">Intercept</td>
<td class="gt_row gt_right" headers="Estimate">−7.574</td>
<td class="gt_row gt_right" headers="Std. Error">1.299</td>
<td class="gt_row gt_right" headers="t value">−5.832</td>
<td class="gt_row gt_left" headers="p value" style="font-weight: bold">&lt; 0.01</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Term">Functional trophic asymmetry</td>
<td class="gt_row gt_right" headers="Estimate">18.380</td>
<td class="gt_row gt_right" headers="Std. Error">1.374</td>
<td class="gt_row gt_right" headers="t value">13.381</td>
<td class="gt_row gt_left" headers="p value" style="font-weight: bold">&lt; 0.01</td>
</tr>
</tbody><tfoot class="gt_sourcenotes">
<tr class="odd">
<td colspan="5" class="gt_sourcenote">Model fit: residual standard error = 5.01 (df = 1240);<br>
R² = 0.126; adjusted R² = 0.126; F(1, 1240) = 179, p &lt; 0.01.</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../Supplementary_material/S1_trait_based_assembly.html" class="pagination-link" aria-label="Trait-based inference of ecological network assembly">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Trait-based inference of ecological network assembly</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../Supplementary_material/S3_historical_biogeography.html" class="pagination-link" aria-label="Climate change drove the deep-time assembly of biodiversity">
        <span class="nav-page-text"><span class="chapter-number">C</span>&nbsp; <span class="chapter-title">Climate change drove the deep-time assembly of biodiversity</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>